wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java

```diff
@@ -293,27 +289,21 @@ public void stop(String workspaceId, Map<String, String> options) throws NotFoun
         }
         eventService.publish(DtoFactory.newDto(WorkspaceStatusEvent.class)
                                        .withWorkspaceId(workspaceId)
-                                       .withPrevStatus(WorkspaceStatus.RUNNING)
-                                       .withStatus(WorkspaceStatus.STOPPING));
+                                       .withPrevStatus(state.status)
+                                       .withStatus(STOPPING));
 
         try {
             state.runtime.stop(options);
```

Comment:
I think that it's a bit overhead to do the async stop in `WorkspaceRuntimes` because we do the async call in `WorkspaceManager`.